### ISSUE TITLE
SDXL implementation? 

### ISSUE THREAD

**Issue by DoctorDerp** [aria1th/Hypernetwork-MonkeyPatch-Extension#55]
Just wondering if you will be making an SDXL implementation for the extension, thanks.

**Comment by aria1th** [comment]
@DoctorDerp I'm still waiting for [dev](https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/dev) branch to merge - since it will be *big* changes for the extension!

**Comment by DoctorDerp** [comment]
> @DoctorDerp I'm still waiting for [dev](https://github.com/AUTOMATIC1111/stable-diffusion-webui/tree/dev) branch to merge - since it will be _big_ changes for the extension!

It seems automatic will be removing hypernetwork training from future webui builds https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/11857#discussioncomment-6480804 in lieu of extensions.  So I am just wondering what the future for hypernetwork training and SDXL looks like.  Does that mean you will bear the mantle of that implementation or is hypernetwork training for sdxl dead? man that would be a bummer....

**Comment by aria1th** [comment]
@DoctorDerp I agree with that - training scripts should be separated as extension (it was mentioned several times)

Since hypernetwork is just simple cross-attention focused adapter module, it will be one of the 'additional networks'.

It does not mean there won't be hypernetwork support, it should still work, but training will be separated - I would maintain it with cleaup code.

But unfortunately, due to recent gradio updates, the changes are too aggressive, so I have to give up backward compatibility (pre-1.5.1) . This is not due to XL - just by gradio... and most of the extensions are having similar problems.

I'll do my best - including some simple training extensions for LoRA.

**Comment by DoctorDerp** [comment]
If Hypernetwork training for SDXL can be implemented in any way I am all for it no matter what form it takes.  Extensions or not.  I've tried so many times to reproduce the quality of results I get from my Hypernetworks (custom training parameters) with LORA, but they never could match the Hypernetworks quality, speed, cohesion and flexibility.  So I have been sold on Hypernetworks capabilities over other methods for a long time now.  I been itching to make more custom hypernetworks for the community on the SDXL, but alas its not implemented so here we are. If you had to guess how long do you think till Hypernetwork training is implemented for SDXL?  Also, just wanted to say thanks for your work on the existing extension.
